xonsh/completers/path.py

```diff
@@ -151,6 +151,59 @@ def _quote_paths(paths, start, end):
     return out
 
 
+def joinpath(path):
+    if path is None:
+        return ''
+    if len(path) == 0:
+        return ''
+    if path == ('',):
+        return get_sep()
+    elif path[0] == '':
+        return get_sep() + _normpath(os.path.join(*path))
+    else:
+        return _normpath(os.path.join(*path))
+
+
+def splitpath(path):
+    path = _normpath(path)
+    if path.startswith(get_sep()):
+        pre = ('', )
+    else:
+        pre = ()
+    return pre + _splitpath(path, ())
+
+
+def _splitpath(path, sofar=()):
+    folder, path = os.path.split(path)
+    if path == "":
+        return sofar[::-1]
+    elif folder == "":
+        return (sofar + (path, ))[::-1]
+    else:
+        return _splitpath(folder, sofar + (path, ))
+
+
+def expanded_match(ref, typed):
```

Comment:
> and maybe difflib would give some better results.

Yeah... No:

```
(rootenv)~/dev/ipython master$ SequenceMatcher(None, '-aaa','aaaa').ratio()
0.75
(rootenv)~/dev/ipython master$ SequenceMatcher(None, 'a-aa','aaaa').ratio()
0.5
(rootenv)~/dev/ipython master$ SequenceMatcher(None, 'aa-a','aaaa').ratio()
0.75
(rootenv)~/dev/ipython master$ SequenceMatcher(None, 'aaa-','aaaa').ratio()
0.75
```

extend with more `a`s and move the dash:

![dsffi6roes0aaaaa 0asuvork5cyii](https://cloud.githubusercontent.com/assets/335567/15904365/9242dc56-2d65-11e6-9309-2354bf432068.png)